server/src/main/java/org/opensearch/index/store/remote/filecache/FileCacheFactory.java

```diff
@@ -37,25 +40,52 @@
  */
 public class FileCacheFactory {
 
-    public static FileCache createConcurrentLRUFileCache(long capacity, CircuitBreaker circuitBreaker) {
-        return new FileCache(createDefaultBuilder().capacity(capacity).build(), circuitBreaker);
+    public static FileCache createConcurrentLRUFileCache(long capacity, CircuitBreaker circuitBreaker, FileTrackerImp fileTrackerImp) {
+        return createFileCache(createDefaultBuilder(fileTrackerImp).capacity(capacity).build(), circuitBreaker);
     }
 
-    public static FileCache createConcurrentLRUFileCache(long capacity, int concurrencyLevel, CircuitBreaker circuitBreaker) {
-        return new FileCache(createDefaultBuilder().capacity(capacity).concurrencyLevel(concurrencyLevel).build(), circuitBreaker);
+    public static FileCache createConcurrentLRUFileCache(
+        long capacity,
+        int concurrencyLevel,
+        CircuitBreaker circuitBreaker,
+        FileTrackerImp fileTrackerImp
```

Comment:
I think it needs to be the `FileTracker` interface instead of the implementation?